### PR TITLE
Chore(#1): add workflow

### LINKED ISSUE #1: Add github workflow
Suggested Jobs:
- setup;
- check-lint;
- run-tests;
- build;

### PR DESCRIPTION
**Pull request by mari1912**
## Description

Add some jobs to check the setup, lint and building for frontend and backend.

## Related Issues
- Closes #1 

## Progress

I did some tests in another repo and it's working fine, but I will try to create another branch based on this one to check if the checks run

### CODE REVIEW

**Comment by lfjnascimento** [comment]
Can you check the errors that appear in the [actions](https://github.com/profusion/kernelCI-dashboard/actions) ?

**Comment by mari1912** [comment]
> Can you check the errors that appear in the [actions](https://github.com/profusion/kernelCI-dashboard/actions) ?

Oh, my bad! It's fixed now

File: dashboard/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import React from 'react';
 import './button.css';
```
Comment (lfjnascimento) [inline dashboard/src/components/Button/Button.tsx:1]: What is the reason behind this? I don't think we need to worry about this component, it's just for testing the storybook so we can remove it after our first component is done.

**Review by lfjnascimento — APPROVED** [review]

File: dashboard/package.json [hunk 2]
```diff
@@ -6,10 +6,10 @@
   "scripts": {
     "dev": "vite",
     "build": "tsc -b && vite build",
-    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview",
     "storybook": "storybook dev -p 6006",
-    "build-storybook": "storybook build"
+    "build-storybook": "storybook build",
+    "lint": "eslint  **/*.tsx **/*.ts --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview"
```
Comment (lfjnascimento) [inline dashboard/package.json:12]: I think it would be better organized if we put this change in its own commit and explain why

File: dashboard/src/components/Button/Button.tsx (hunk 1 above)
Comment (mari1912) [reply dashboard/src/components/Button/Button.tsx:1]: i had to change this because of the build workflow, it was failing because it says the React import was not being used. we still need to define some linter rules in the application

File: dashboard/package.json (hunk 2 above)
Comment (mari1912) [reply dashboard/package.json:12]: the `lint` command was broken, because we don't have a eslint config file, I though it was better to define the linter rules in another pull request
